Content.Server/Ensnaring/Components/EnsnaringComponent.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Threading;
+using Content.Shared.Ensnaring.Components;
+
+namespace Content.Server.Ensnaring.Components;
+[RegisterComponent]
+[ComponentReference(typeof(SharedEnsnaringComponent))]
+public sealed class EnsnaringComponent : SharedEnsnaringComponent
+{
+    /// <summary>
+    /// Should movement cancel breaking out?
+    /// </summary>
+    [ViewVariables(VVAccess.ReadWrite)]
+    [DataField("canMoveBreakout")]
+    public bool CanMoveBreakout;
+
+    public CancellationTokenSource? CancelToken;
+}
+
+/// <summary>
+/// Used for the do after event to free the entity that owns the <see cref="EnsnareableComponent"/>
+/// </summary>
+public sealed class FreeEnsnareDoAfterComplete : EntityEventArgs
+{
+    public readonly EnsnaringComponent? EnsnaringComponent;
+
+    public FreeEnsnareDoAfterComplete(EnsnaringComponent ensnaringComponent)
+    {
+        EnsnaringComponent = ensnaringComponent;
+    }
+}
+
+/// <summary>
+/// Used for the do after event when it fails to free the entity that owns the <see cref="EnsnareableComponent"/>
+/// </summary>
+public sealed class FreeEnsnareDoAfterCancel : EntityEventArgs
+{
+    public readonly EnsnaringComponent? EnsnaringComponent;
```

Comment:
You should really use the EntityUid instead as I don't think the component lifetime is handled, then just do a TryComp<T> when the do_after completes.